website/project/tasks.py

```diff
@@ -44,7 +44,7 @@ def update_collecting_metadata(node, saved_fields):
         if node.is_public:
             update_collected_metadata(node._id)
         else:
-            if 'is_public' in saved_fields:
+            if set(saved_fields).intersection({'spam_status', 'is_deleted', 'deleted', 'is_public'}):
```

Comment:
This doesn't seem to change behavior for making a collected node private, which was the concern. Are there any tests for making a collected node private and hitting the search endpoint to ensure it doesn't show?